source/codegen/format-mi-metadata.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import codecs
+import fnmatch
+import os
+import sys
+from enum import Enum
+from pprint import pprint
+from typing import Set
+from unittest import FunctionTestCase
+
+from template_helpers import load_metadata
+
+
+class Formatter(object):
```

Comment:
Note: I took this Formatter class straight from hapigen: see [here](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/hapigen/hapigen/source/hapigen_utilities/hapigen_utilities/Formatter.py)